src/pe/resource.rs

```diff
@@ -0,0 +1,1942 @@
+    }
+
+    /// Stringize the [`StringFileInfo::product_name`] slice into a [`String`].
+    pub fn product_name(&self) -> Option<String> {
+        self.product_name.and_then(to_utf16_string)
+    }
+
+    /// Stringize the [`StringFileInfo::product_version`] slice into a [`String`].
+    pub fn product_version(&self) -> Option<String> {
+        self.product_version.and_then(to_utf16_string)
+    }
+
+    /// Stringize the [`StringFileInfo::special_build`] slice into a [`String`].
+    pub fn special_build(&self) -> Option<String> {
+        self.special_build.and_then(to_utf16_string)
+    }
+}
+
+/// Represents a version information
+#[derive(Copy, Clone)]
+pub struct VersionInfo<'a> {
+    /// Raw data of entire [`RT_VERSION`] area.
+    data: &'a [u8],
+    /// Fixed file information.
+    pub fixed_info: Option<VsFixedFileInfo>,
+    /// Dynamic key-value file information.
+    pub string_info: StringFileInfo<'a>,
+}
+
+impl fmt::Debug for VersionInfo<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("VersionInfo")
+            .field(
+                "data",
+                &format_args!("{:02x?} ({} bytes)", &self.data, self.data.len()),
+            )
+            .field("fixed_info", &self.fixed_info)
+            .field("string_info", &self.string_info)
+            .finish()
+    }
+}
+
+impl<'a> VersionInfo<'a> {
+    pub fn parse(
+        pe: &'a [u8],
+        bytes: &'a [u8],
+        it: ResourceEntryIterator<'a>,
+        sections: &[section_table::SectionTable],
+        file_alignment: u32,
+        opts: &options::ParseOptions,
+    ) -> error::Result<Option<Self>> {
+        if let Some(entry) = it.find_by_id(RT_VERSION)? {
+            let offset_to_data =
+                match entry.recursive_next_depth(bytes, |e| e.offset_to_data().is_none())? {
+                    Some(next) => match next.offset_to_data() {
+                        Some(offset_to_data) => offset_to_data,
+                        None => return Ok(None),
+                    },
+                    None => return Ok(None),
+                };
+            let mut offset = offset_to_data as usize;
+            let data_entry = bytes.gread_with::<ResourceDataEntry>(&mut offset, scroll::LE)?;
+            let rva = data_entry.offset_to_data as usize;
+            offset = utils::find_offset(rva, sections, file_alignment, opts).ok_or_else(|| {
+                error::Error::Malformed(format!(
+                    "Cannot map ResourceDataEntry rva {:#x} into offset",
+                    rva
+                ))
+            })?;
+
+            let data = pe
+                .pread_with::<&[u8]>(offset, data_entry.size as usize)
+                .map_err(|_| {
+                    error::Error::Malformed(format!("Cannot map RT_VERSION at {offset:#x}"))
+                })?;
+            let iterator = ResourceStringIterator { data };
+            let strings = iterator.collect::<Result<Vec<_>, _>>()?;
```

Comment:
perf: i'm not sure you have to allocate here if you just use it to generate fixed_info?

Reply:
Unfortunately yes, we need it to find a `VS_VERSION_INFO` entry.